lightning/src/ln/channelmanager.rs

```diff
@@ -2825,6 +2842,7 @@ impl<ChanSigner: ChannelKeys> ChannelMessageHandler for ChannelManager<ChanSigne
 				log_debug!(self, "Marking channels with {} disconnected and generating channel_updates", log_pubkey!(their_node_id));
 				channel_state.by_id.retain(|_, chan| {
 					if chan.get_their_node_id() == *their_node_id {
+						no_channels_remain = false;
```

Comment:
Right, I think. Because the peer can open a channel with us at any point while connected, we store their features in per_peer_state as long as they're connected. If we have a channel with them, we want to keep the entry around (cause eventually we should be storing the channels in per_peer_state too).

Reply:
Okay, so if it's so I think it's buggy. You should have no_channels_remain = false if channel_state.len() > 0 after the retain. Because if every channel is already is_shutdown(), we should prune this peer out.

Reply:
Oops, good catch!